alkira/resource_alkira_connector_azure_vnet_helper.go

```diff
@@ -0,0 +1,97 @@
+package alkira
+
+import (
+	"github.com/alkiranet/alkira-client-go/alkira"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// constructVnetRouting construct connector_azure_vnet routing options
+func constructVnetRouting(d *schema.ResourceData) (*alkira.ConnectorVnetRouting, error) {
+
+	importOptions := alkira.ConnectorVnetImportOptions{}
+	importOptions.RouteImportMode = d.Get("routing_options").(string)
+	importOptions.PrefixListIds = convertTypeListToIntList(d.Get("routing_prefix_list_ids").([]interface{}))
+
+	serviceRoutes := alkira.ConnectorVnetServiceRoutes{}
+
+	// Processing vnet_subnet blocks
+	for _, block := range d.Get("vnet_subnet").(*schema.Set).List() {
+		content := block.(map[string]interface{})
+
+		if _, ok := content["routing_options"].(string); ok {
+			subnetImportOption := alkira.ConnectorVnetImportOptionsSubnet{}
+
+			if v, ok := content["subnet_id"].(string); ok {
```

Comment:
Just thinking out loud here. I found out that if `content["routing_options"]` is not a string type ok evaluates to false. We are not notified that there was a failure in our type assertion. The key could exist and its associated value, but the program continues to run as if it did not. I've been thinking about this and researching and I haven't found a better solution. Something to think about for us. This behavior caused a bug for me that testing revealed.

Reply:
Yep, so for now, testing is the best we could do to catch this kinds of case… so I made bunch of new configs.

Even it failed, API should have the proper response no matter what….